checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.hadoop.fs.FileSystem;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.SparkConf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import scala.Serializable;
+import scala.Tuple2;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A spark job to test the integration of Spark with Alluxio. This class will be triggered
+ * in yarn-client and yarn-cluster mode through spark-submit.
+ *
+ * This job requires a running Spark cluster, but not requires a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized in Spark
+ * driver and executors.
+ */
+public class SparkIntegrationChecker implements Serializable{
+
+  private static final long serialVersionUID = 1L;
+
+  @Parameter(names = {"-p", "--partition"}, description = "The user defined partition number, "
+      + "by default the value is 10")
+  private int mPartitions;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkIntegrationChecker.class);
+  private List<Tuple2<Status, String>> mSparkJobResult;
+  private boolean mAlluxioHAMode = false;
+  private String mZookeeperAddress = "";
+
+  /* The Spark driver and executors performIntegrationChecks results*/
+  private enum Status {
+    FAILCLASS, // Spark driver or executors cannot recognize Alluxio classes
+    FAILFS,    // Spark driver or executors cannot recognize Alluxio filesystem
+    FAILHA,    // Spark driver cannot support Alluxio-HA mode
+    SUCCESS;
+  }
+
+  /**
+   * Implements check integration of Spark with Alluxio.
+   *
+   * @param sc current JavaSparkContext
+   * @return Status performIntegrationChecks results
+   */
+  private Status run(JavaSparkContext sc) {
+    // Checks whether Spark driver can recognize Alluxio classes and filesystem
+    Status driverStatus = performIntegrationChecks();
+    if (driverStatus.equals(Status.FAILCLASS)) {
+      LOG.error("Spark driver: " + getAddress() + " failed to recognize Alluxio classes.");
+      return driverStatus;
+    } else if (driverStatus.equals(Status.FAILFS)) {
+      LOG.error("Spark driver: " + getAddress() + " failed to recognize Alluxio filesystem.");
+      return driverStatus;
+    } else {
+      LOG.info("Spark driver: " + getAddress() + " can recognize Alluxio filesystem.");
+    }
+
+    // Supports Alluxio high availability mode
+    if (alluxio.Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
+      mAlluxioHAMode = true;
+      if (!alluxio.Configuration.containsKey(PropertyKey.ZOOKEEPER_ADDRESS)) {
+        LOG.error("Zookeeper address has not been set:");
```

Comment:
typo "Zookeeper address has not been set:" -> "Zookeeper address has not been set."